homeassistant/components/drop_connect/sensor.py

```diff
@@ -0,0 +1,272 @@
+"""Support for DROP sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfVolume
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    CONF_COORDINATOR,
+    CONF_DEVICE_TYPE,
+    DEV_FILTER,
+    DEV_HUB,
+    DEV_LEAK_DETECTOR,
+    DEV_PROTECTION_VALVE,
+    DEV_PUMP_CONTROLLER,
+    DEV_RO_FILTER,
+    DEV_SOFTENER,
+    DOMAIN,
+)
+from .coordinator import DROPDeviceDataUpdateCoordinator
+from .entity import DROPEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+FLOW_ICON = "mdi:shower-head"
+GAUGE_ICON = "mdi:gauge"
+TDS_ICON = "mdi:water-opacity"
+
+
+@dataclass(kw_only=True)
+class DROPSensorEntityDescription(SensorEntityDescription):
+    """Describes DROP sensor entity."""
+
+    value_fn: Callable[[DROPDeviceDataUpdateCoordinator], float | int | None]
+
+
+SENSORS: dict[str, DROPSensorEntityDescription] = {
+    "current_flow_rate": DROPSensorEntityDescription(
+        key="current_flow_rate",
+        icon="mdi:shower-head",
+        native_unit_of_measurement="gpm",
+        suggested_display_precision=1,
+        value_fn=lambda device: device.current_flow_rate,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    "peak_flow_rate": DROPSensorEntityDescription(
+        key="peak_flow_rate",
+        icon="mdi:shower-head",
+        native_unit_of_measurement="gpm",
+        suggested_display_precision=1,
+        value_fn=lambda device: device.peak_flow_rate,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    "water_used_today": DROPSensorEntityDescription(
+        key="water_used_today",
+        device_class=SensorDeviceClass.WATER,
+        native_unit_of_measurement=UnitOfVolume.GALLONS,
+        suggested_display_precision=1,
+        value_fn=lambda device: device.water_used_today,
+    ),
+    "average_water_used": DROPSensorEntityDescription(
+        key="average_water_used",
+        device_class=SensorDeviceClass.WATER,
+        native_unit_of_measurement=UnitOfVolume.GALLONS,
+        suggested_display_precision=0,
+        value_fn=lambda device: device.average_water_used,
+    ),
+    "capacity_remaining": DROPSensorEntityDescription(
+        key="capacity_remaining",
+        device_class=SensorDeviceClass.WATER,
```

Comment:
Also, you can combine both state_class and device_class. Maybe you don't have areason to do so, but wanted to let you know

Reply:
The definitions that use state_class don't have an appropriate device_class that I've found. In particular, the flow rates (GPM) and TDS (total dissolved solids in water). It's possible that I've overlooked something or am misunderstanding your suggestion, however...